hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutBackupOperation.java

```diff
@@ -75,8 +86,30 @@ public void afterRun() throws Exception {
         if (recordInfo != null) {
             evict(true);
         }
+        final MapServiceContext mapServiceContext = mapService.getMapServiceContext();
+        final MapEventPublisher mapEventPublisher = mapServiceContext.getMapEventPublisher();
+        if (!wanOriginated) {
+            publishWANReplicationEventBackup(mapServiceContext, mapEventPublisher);
+        }
+
     }
 
+    private void publishWANReplicationEventBackup(MapServiceContext mapServiceContext, MapEventPublisher mapEventPublisher) {
+        if (mapContainer.getWanReplicationPublisher() == null || mapContainer.getWanMergePolicy() == null) {
```

Comment:
We have a duplication of the logic. If you add a method 'isWanReplicationEnabled' to the MapContainer; the problem is solved. As soon as you copy paste code, you need to think about pulling out a method.